plugin/src/main/java/org/openrewrite/gradle/RewritePlugin.java

```diff
@@ -98,10 +100,16 @@ private static void configureProject(Project project, RewriteExtension extension
             }
 
             //Collect Java metadata for each project (used for Java Provenance)
-            //Using the older javaConvention because we need to support older versions of gradle.
-            @SuppressWarnings("deprecation")
-            JavaPluginConvention javaConvention = project.getConvention().getPlugin(JavaPluginConvention.class);
-            javaConvention.getSourceSets().all(sourceSet -> {
+            SourceSetContainer sourceSets;
+            if (project.getGradle().getGradleVersion().compareTo("7.0") > 0) {
```

Comment:
```suggestion
            if (GradleVersion.current().compareTo(GradleVersion.version("7.1")) >= 0) {
```

Specifically, the `GradleVersion` has better semver semantics and there would have been bugs introduced with the previous implementation for 7.0.1 and 7.0.2 as the `JavaPluginExtension#getSourceSets()` method is first introduced in 7.1.